src/Microsoft.HttpRepl/Commands/TreeNode.cs

```diff
@@ -21,9 +21,9 @@ public TreeNode(Formatter formatter, string prefix, string entry)
 
         private TreeNode(Formatter formatter, string prefix, string entry, int depth)
         {
-            _formatter = formatter;
+            _formatter = formatter ?? throw new ArgumentNullException(nameof(formatter));
             formatter.RegisterEntry(prefix.Length, depth);
```

Comment:
You haven't checked prefix for null before using `prefix.Length` here.

(also implied: there's a missing unit test)

Reply:
Checked, added tests and re-ordered the checks too.